src/display/annotation_layer.js

```diff
@@ -2509,7 +2509,20 @@ class FileAttachmentAnnotationElement extends AnnotationElement {
   render() {
     this.container.className = "fileAttachmentAnnotation";
 
-    const trigger = document.createElement("div");
+    let trigger;
+    if (this.data.hasAppearance) {
+      trigger = document.createElement("div");
+    } else {
+      // Unfortunately it seems that it's not clearly specified exactly what
```

Comment:
Can we just use the background-image css property and plug it on a new attribute we could add to the div node ?

Reply:
Unfortunately that doesn't work with the ref-tests (and I don't really have time to try and improve that), and it also doesn't auto-size the icons (meaning they overflow and are clipped) which the `img` approach just handles.

Reply:
Ah yes it's likely the same issue we've with the missing blue background we don't have for some tests with text fields.

Reply:
> Ah yes it's likely the same issue we've with the missing blue background we don't have for some tests with text fields.

Aren't those ones actually *intermittent* failures though, even is some of them are happening fairly commonly?

Reply:
My impress is that we've the blue background when we're making ref but not when we're just comparing...
I had a look at some point and if iirc the generated svg is correct but for any reason the image dumper is wrong.
Yes they look like intermittent, but for any reason each time we touch the annotation stuff, I've the impress that they are failing...
Anyway, the blue background is rendered thanks to an image (it's why I made the relation with what you said):
 - https://github.com/mozilla/pdf.js/blob/master/web/annotation_layer_builder.css#L17
 - https://github.com/mozilla/pdf.js/blob/master/web/annotation_layer_builder.css#L91